CIP-0022/CIP-0022.md

```diff
@@ -0,0 +1,134 @@
+---
+CIP: 22
+Title: Pool operator verification
+Authors: Andrew Westberg <andrewwestberg@gmail.com>, Martin Lang <martin@martinlang.at>, Ola Ahlman <ola@ahlnet.nu>
+Comments-URI: no comments yet
+Status: Active
+Type: Informational
+Created: 2021-06-21
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This proposal describes a method allowing a stakepool operator to provide credentials to verify that they are the rightful manager for their stakepool.
+
+# Motivation
+
+Many websites such pooltool.io, adapools.org, and others need to allow pool operators special access to modify the way their pool appears on the website. SPOCRA and other organizations also have a need to allow voting on proposals and ensure that each vote cast is from a valid pool operator. Today, these sites and organizations all use different techniques for validating pool operators.
+
+pooltool.io - Validates operators by receiving 1 ada spent from the pool's registered rewards account
+
+adapools.org - Validates operators by requesting that the operator include a special generated value in their extended pool metadata json file.
+
+This proposal desires to simplify and streamline a single approach that all can reference in order to verify that a pool operator is who they say they are.
+
+Current Tools that have implemented it already or have plans to implement it:
+* [CNCLI](https://github.com/AndrewWestberg/cncli)
+* [JorManager](https://bitbucket.org/muamw10/jormanager/)
+* [StakePoolOperator Scripts](https://github.com/gitmachtl/scripts)
+* [CNTools](https://cardano-community.github.io/guild-operators/#/Scripts/cntools)
+
+
+# Specification
+
+In order to achieve the goals of this CIP, the pool operator needs to provide some credential or credentials to the validating party which cannot be spoofed. The VRF pool keys and VRF signature algorithm implemented in libsodium are chosen to build and provide this credential/signature. This signature can then be validated and the operator verified without ever exposing any of the pool's private key information. This technique is very similar to verifying that a block produced by another pool is valid. The only difference is that instead of validating the slot seed for a given pool, we're validating a pre-determined message hash.
+
+## Verification Steps:
+
+1. Stakepool Operator (SPO) sends their pool_id and public pool.vrf.vkey to the Validating Server (VS)
+2. VS validates that the vrf hash in the pool's registration certificate on the blockchain matches the blake2b hash of the sent vkey.
```

Comment:
In this section I think it is worth pointing out clearly that the VRF key can be changed at any time by the SPO when they update their pool registration (without changing their pool identity). The VRF is a "hot" key.

So implementations following these steps (including point 2) will be able to check that the VRF was the right one at a single point in time. That's likely to be enough for most applications. I think it's worth pointing out here however as something to be aware of for some applications.

Reply:
@dcoutts I've added language to ensure that the VS uses the latest regcert on the chain. Also added language that the VRF key is hot and this is a point-in-time verfication.